script.js

```diff
@@ -1,4 +1,387 @@
+
+      // If rank is 1, 11, 12, or 13, set cardName to the ace or face card's name
+      // Blackjack - all picture cards are now rank 10
+      if (cardName == 1) {
+        cardName = "ace";
+      } else if (cardName == 11) {
+        cardName = "jack";
+        rankCounter = 10;
+      } else if (cardName == 12) {
+        cardName = "queen";
+        rankCounter = 10;
+      } else if (cardName == 13) {
+        cardName = "king";
+        rankCounter = 10;
+      }
+
+      // Create a new card with the current name, suit, and rank
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+      };
+
+      // Add the new card to the deck
+      cardDeck.push(card);
+
+      // Increment cardCounter to iterate over the next rank
+      cardCounter += 1;
+    }
+
+    // Increment the suit index to iterate over the next suit
+    suitIndex += 1;
+  }
+
+  // Return the completed card deck
+  return cardDeck;
+};
+
+// In-class exercise
+var convertSuitToEmoji = function (suit) {
+  if (suit == "spades") {
+    return "♠️";
+  }
+  if (suit == "hearts") {
+    return "♥️";
+  }
+  if (suit == "clubs") {
+    return "♣️";
+  }
+  if (suit == "diamonds") {
+    return "♦️";
+  }
+  // If we reach here, we entered an invalid suit
+  return "Invalid Suit!";
+};
+
+var convertCardNameToEmoji = function (cardName) {
+  if (cardName == 2) {
+    return "2️⃣";
+  }
+  if (cardName == 3) {
+    return "3️⃣";
+  }
+  if (cardName == 4) {
+    return "4️⃣";
+  }
+  if (cardName == 5) {
+    return "5️⃣";
+  }
+  if (cardName == 6) {
+    return "6️⃣";
+  }
+  if (cardName == 7) {
+    return "7️⃣";
+  }
+  if (cardName == 8) {
+    return "8️⃣";
+  }
+  if (cardName == 9) {
+    return "9️⃣";
+  }
+  if (cardName == 10) {
+    return "🔟";
+  }
+  return cardName;
+};
+
+// Initialise the card deck representation as an array of objects
+// Modify template code to generate deck with function call
+var deck = makeDeck();
+
+// Shuffle the deck and save it in a new variable shuffledDeck
+// to communicate that we have shuffled the deck.
+var shuffledDeck = shuffleCards(deck);
+
+// Global variables
+// Winning value, minimum value to hit, player's decision, game end, list to store player and dealer hand
+var BLACKJACK = 21;
+var dealerHitMin = 16;
+var playerDecision = false;
+var endGame = false;
+var playerHand = [];
+var dealerHand = [];
+
+// Comfortable Image
+var winImage = '<img src="https://c.tenor.com/Hj2-u4VELREAAAAi/655.gif"/>';
+var loseImage =
+  '<img src="https://c.tenor.com/WPvUD2aORooAAAAC/we-lost-losing.gif"/>';
+var tieImage =
+  '<img src="https://c.tenor.com/wyfhYqF1tJIAAAAC/mark-wahlberg-wahlberg.gif"/>';
+var dealerWinImage =
+  '<img src= "https://c.tenor.com/C7CZoUiIIUQAAAAC/poker-cards-deal-with-it.gif"/>';
+var addWelcomeImage = document.getElementById("output-div");
+addWelcomeImage.innerHTML +=
+  '<img src="https://c.tenor.com/Phz7IULNI9QAAAAC/box-hi.gif">';
```

Comment:
These strings don't change value throughout the whole program, so they are constants and the naming convention for constants is `WIN_IMAGE`